securedrop/journalist_app/api.py

```diff
@@ -134,41 +134,43 @@ def get_token():
 
     @api.route('/sources', methods=['GET'])
     @token_required
-    def get_all_sources():
+    def get_all_sources() -> Tuple[flask.Response, int]:
         sources = Source.query.filter_by(pending=False, deleted_at=None).all()
         return jsonify(
             {'sources': [source.to_json() for source in sources]}), 200
 
     @api.route('/sources/<source_uuid>', methods=['GET', 'DELETE'])
     @token_required
-    def single_source(source_uuid):
+    def single_source(source_uuid: str) -> Tuple[flask.Response, int]:
         if request.method == 'GET':
             source = get_or_404(Source, source_uuid, column=Source.uuid)
             return jsonify(source.to_json()), 200
         elif request.method == 'DELETE':
             source = get_or_404(Source, source_uuid, column=Source.uuid)
             utils.delete_collection(source.filesystem_id)
             return jsonify({'message': 'Source and submissions deleted'}), 200
+        else:
+            abort(404)
```

Comment:
This is a good addition, as Flask's implicit support for `HEAD` means we're setting up a 500 error when we fall out of the `if` without handling it, but I think a 405 (method not allowed) status would be more accurate here and the other places you've caught this.